diff --git a/vswitchd/vswitch.ovsschema b/vswitchd/vswitch.ovsschema
index 4873cfde7..5f42a27ec 100644
--- a/vswitchd/vswitch.ovsschema
+++ b/vswitchd/vswitch.ovsschema
@@ -1,6 +1,6 @@
 {"name": "Open_vSwitch",
  "version": "8.3.0",
- "cksum": "3781850481 26690",
+ "cksum": "2334774277 27445",
  "tables": {
    "Open_vSwitch": {
      "columns": {
@@ -13,6 +13,10 @@
          "type": {"key": {"type": "uuid",
                           "refTable": "Bridge"},
                   "min": 0, "max": "unlimited"}},
+       "p4_devices": {
+         "type": {"key": {"type": "uuid",
+                          "refTable": "P4_Device"},
+                  "min": 0, "max": "unlimited"}},
        "manager_options": {
          "type": {"key": {"type": "uuid",
                           "refTable": "Manager"},
@@ -717,4 +721,20 @@
                   "value": {"type": "integer",
                           "minInteger": 0,
                           "maxInteger": 4095},
-                  "min": 0, "max": "unlimited"}}}}}}
+                  "min": 0, "max": "unlimited"}}}},
+   "P4_Device": {
+     "columns": {
+       "device_id": {
+         "type": {"key": {"type": "integer",
+                          "minInteger": 1,
+                          "maxInteger": 16},
+                  "min": 0, "max": 1}},
+       "config_file_path": {
+         "type": {"key": {"type": "string"},
+                  "min": 0, "max": 1}},
+       "bridges": {
+         "type": {"key": {"type": "uuid",
+                          "refTable": "Bridge",
+                          "refType": "weak"},
+                  "min": 0, "max": "unlimited"}}},
+       "indexes": [["device_id"]]}}}
diff --git a/vswitchd/vswitch.xml b/vswitchd/vswitch.xml
index 026b5e2ca..c67cfeca1 100644
--- a/vswitchd/vswitch.xml
+++ b/vswitchd/vswitch.xml
@@ -63,6 +63,10 @@
         Set of bridges managed by the daemon.
       </column>
 
+      <column name="p4_devices">
+        Set of p4_devices managed by the daemon.
+      </column>
+
       <column name="ssl">
         SSL used globally by the daemon.
       </column>
@@ -6765,4 +6769,25 @@ ovs-vsctl add-port br0 p0 -- set Interface p0 type=patch options:peer=p1 \
       id.
     </column>
   </table>
+
+  <table name="P4_Device">
+    <p>
+      Store p4 device specific information. This is shared between runtime
+      clients and the vswitchd.
+    </p>
+
+    <column name="device_id">
+      Identifies the p4 device_id which corresponds to pipeline in target.
+    </column>
+
+    <column name="config_file_path">
+      Identifies the config file associated with the device_id.
+      This information is consumed by target.
+    </column>
+
+    <column name="bridges">
+      Software bridges associated with the P4 device.
+    </column>
+  </table>
+
 </database>
